scipy/stats/_hypotests.py

```diff
@@ -9,11 +9,98 @@
 from scipy.special import gamma, kv, gammaln
 from . import _wilcoxon_data
 import scipy.stats
+import scipy.stats.stats
+from functools import wraps
+
+
+# although _vectorize_2s_hypotest_factorycan be generalized to n-d,
+# it might make some sense to have a separate decorator for the 1d case
+# because it can be quite a bit simpler - we can use np.apply_along_axis,
+# for instance
+def _vectorize_1s_hypotest_factory(result_creator, default_axis=0):
+    def vectorize_hypotest_decorator(hypotest_fun_in):
+        @wraps(hypotest_fun_in)
+        def vectorize_hypotest_wrapper(x, *args, axis=default_axis,
+                                       nan_policy='propagate', **kwds):
+
+            x = np.atleast_1d(x)
+
+            # Addresses nan_policy="raise"
+            _, nan_policy = scipy.stats.stats._contains_nan(x, nan_policy)
+
+            # Addresses nan_policy="omit"
+            # Assumes nan_policy="propagate" is handled hypotest_fun_in
+            def hypotest_fun(x):
+                if nan_policy == 'omit':
+                    x = x[~np.isnan(x)]
+                return hypotest_fun_in(x, *args, **kwds)
+
+            if axis is None:
+                x = x.ravel()
+                return hypotest_fun(x)
+            else:
+                x = np.moveaxis(x, axis, -1)
+                res = np.apply_along_axis(hypotest_fun, axis=-1, arr=x)
+                return result_creator(res)
+
+        return vectorize_hypotest_wrapper
+    return vectorize_hypotest_decorator
+
+
+# this can be generalized to n-d, and it should use the
+# approach for the "omit" nan_policy used above
```

Comment:
I'm going to try refactoring this to use the _vectorize_2s_hypotest_factory. The samples would be broadcast-concatenated so they look like one sample, then split apart to be passed into the base function. Sounds complicated, but I think it will actually be a lot cleaner overall.